src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs

```diff
@@ -30,19 +32,39 @@ public sealed class UnmanagedPSEntry
         /// <param name="argc">
         /// Length of the passed in argument array.
         /// </param>
+        [Obsolete("Callers should now use UnmanagedPSEntry.Start(string[])", error: true)]
         public static int Start(string consoleFilePath, [MarshalAs(UnmanagedType.LPArray, ArraySubType = UnmanagedType.LPWStr, SizeParamIndex = 2)]string[] args, int argc)
         {
+            return Start(args);
+        }
+
+        /// <summary>
+        /// Starts managed MSH.
+        /// </summary>
+        /// <param name="args">
+        /// Command line arguments to the managed MSH
+        /// </param>
+        public static int Start(string[] args)
+        {
+            if (args == null)
```

Comment:
nullable is enabled, so `string[] args` means `args` is not null, no?

Reply:
For public we should check input values always. Users can run `Start(null)`.

Reply:
Right. How this should be done when nullable enabled? Use `Start(string?[] args)`?

Reply:
I think we should announce `string?[] args` - elements can be null.
Done.